maas/resource_maas_configuration_test.go

```diff
@@ -0,0 +1,462 @@
+package maas_test
+
+import (
+	"fmt"
+	"terraform-provider-maas/maas/testutils"
+	"testing"
+
+	"terraform-provider-maas/maas"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestNormalizeConfigValue(t *testing.T) {
+	testCases := []struct {
+		apiValue []byte
+		expected string
+		label    string
+	}{
+		{
+			apiValue: []byte{0x6e, 0x75, 0x6c, 0x6c},
+			expected: "",
+			label:    "literal null",
+		},
+		{
+			apiValue: []byte("\"null\""),
+			expected: "null",
+			label:    "string literal null",
+		},
+		{
+			apiValue: []byte("hello world"),
+			expected: "hello world",
+			label:    "unquoted string",
+		},
+		{
+			apiValue: []byte("example.com:514"),
+			expected: "example.com:514",
+			label:    "unquoted string 2",
+		},
+		{
+			apiValue: []byte("\"example.com:514\""),
+			expected: "example.com:514",
+			label:    "quoted string",
+		},
+	}
+
+	for _, testCase := range testCases {
+		t.Run(testCase.label, func(t *testing.T) {
+			res := maas.NormalizeConfigValue(testCase.apiValue)
+			if res != testCase.expected {
+				t.Errorf("expected %q, got %q for case %q", testCase.expected, res, testCase.label)
+			}
+		})
+	}
+}
+
+func TestAccResourceMAASConfiguration_basic(t *testing.T) {
+	// Test all MAAS Settings cases. Set twice to ensure a change is actually made
+	// incase defaults change, and value 2 is selected to be a typical value to
+	// ensure dev environments aren't greatly changed between test runs.
+	testCases := []struct {
+		key    string
+		value1 string
+		value2 string
+	}{
+		{
+			key:    "active_discovery_interval",
+			value1: "21600",
+			value2: "10800",
+		},
+		{
+			key:    "auto_vlan_creation",
+			value1: "false",
+			value2: "true",
+		},
+		{
+			key:    "boot_images_auto_import",
+			value1: "false",
+			value2: "true",
+		},
+		{
+			key:    "boot_images_no_proxy",
+			value1: "true",
+			value2: "false",
+		},
+		{
+			key:    "commissioning_distro_series",
+			value1: "noble",
+			value2: "noble", // Dev environment does not necessarily have multiple distros
```

Comment:
there is no guarantee that noble is available at the MAAS under test. What about using a similar approach to `TF_ACC_BOOT_RESOURCES_OS` for boot resources? The CI should be updated accordingly to provide to acceptance tests the needed environment variable.